pkg/policy/l4.go

```diff
@@ -109,23 +109,20 @@ func (l4 *L4Filter) AllowsAllAtL3() bool {
 // destination addressing/identity information.
 func (l7 L7DataMap) GetRelevantRules(identity *identity.Identity) api.L7Rules {
 	rules := api.L7Rules{}
-	matched := 0
 
 	if identity != nil {
 		for selector, endpointRules := range l7 {
 			if selector.Matches(identity.Labels.LabelArray()) {
-				matched++
 				rules.HTTP = append(rules.HTTP, endpointRules.HTTP...)
```

Comment:
Nit: and not related to your change really, but `GetRelevantRules` is only used for Kafka from what I can see. Correct me if I am wrong. Should we remove the dead code `rules.HTTP = append(rules.HTTP, endpointRules.HTTP...)`?

Reply:
This function is exported and thus can be used anywhere; thus I think we should keep it as is.

Reply:
Sure. But IMO this dead code is simply confusing and might lead to bugs as we add more L7 protocols. I am ok with leaving it, maybe we can make this protocol specific or atleast add a comment.

Reply:
The function operates on a `L7DataMap`. It cannot make assumptions about its callers. It absolutely has to support all L7 protocols that we support regardless of who is currently using it. It is dead code right now for Envoy but only because Envoy has its own version of this function in Envoy specific code.

Reply:
sounds good! Thanks for the clarification.